builder/amazon/common/ami_config.go

```diff
@@ -20,14 +20,17 @@ type AMIConfig struct {
 	AMIEnhancedNetworking   bool              `mapstructure:"enhanced_networking"`
 	AMIForceDeregister      bool              `mapstructure:"force_deregister"`
 	AMIEncryptBootVolume    bool              `mapstructure:"encrypt_boot"`
+	AMISkipRegister         bool              `mapstructure:"skip_register_ami"`
 }
 
 func (c *AMIConfig) Prepare(ctx *interpolate.Context) []error {
 	var errs []error
-	if c.AMIName == "" {
+	if c.AMIName == "" && !c.AMISkipRegister {
```

Comment:
in https://github.com/mitchellh/packer/blob/master/builder/amazon/ebs/builder.go#L96-L103 it looks like we make use of the AMI name and some other config.

I wonder if we should either not run that step if we're skipping registration, or add another flag like `ForceDeregister` (eg `SkipRegister`) to make it a no op.

I generally like letting the steps decide if they're going to run or not, but we're already picking and choosing which steps we run based on config in this PR.